<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ServoControlBundle</class>
 <widget class="QWidget" name="ServoControlBundle">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>140</width>
    <height>100</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>140</horstretch>
    <verstretch>100</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>140</width>
    <height>100</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>140</width>
    <height>100</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>140</width>
     <height>100</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>140</horstretch>
     <verstretch>100</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>140</width>
     <height>100</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>140</width>
     <height>100</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">QGroupBox{
 	border-width: 1px; 
	border-style: solid; 
	border-radius: 5px;
	border-color: rgb(0, 0, 0);

}


QSpinBox {
	background-color: rgb(255, 255, 255);
}

QSlider::groove:vertical {
border: 1px solid #bbb;
background: white;
height: 68px;
border-radius: 4px;
}

QSlider::sub-page:vertical {
background: qlineargradient(x1: 0, y1: 0,    x2: 0, y2: 1,
    stop: 0 #66e, stop: 1 #bbf);
background: qlineargradient(x1: 0, y1: 0.2, x2: 1, y2: 1,
    stop: 0 #bbf, stop: 1 #55f);
border: 1px solid #777;
height: 68px;
border-radius: 4px;
}

QSlider::add-page:vertical {
background: #fff;
border: 1px solid #777;
height: 10px;
border-radius: 4px;
}

QSlider::handle:vertical {
background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
    stop:0 #eee, stop:1 #ccc);
border: 1px solid #777;
width: 13px;
height:10px;
margin-top: -2px;
margin-bottom: -2px;
border-radius: 4px;
}

QSlider::handle:vertical:hover {
background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
    stop:0 #fff, stop:1 #ddd);
border: 1px solid #444;
border-radius: 4px;
}

QSlider::sub-page:vertical:disabled {
background: #bbb;
border-color: #999;
}

QSlider::add-page:vertical:disabled {
background: #eee;
border-color: #999;
}

QSlider::handle:vertical:disabled {
background: #eee;
border: 1px solid #aaa;
border-radius: 4px;
}

QPushButton {
color: #222;
border: 2px solid #555;
border-radius: 11px;
padding: 5px;
background: qradialgradient(cx: 0.3, cy: -0.4,
fx: 0.3, fy: -0.4,
radius: 1.35, stop: 0 #fff, stop: 1 #BBB);
min-width: 80px;
}

QPushButton:hover {
background: qradialgradient(cx: 0.3, cy: -0.4,
fx: 0.3, fy: -0.4,
radius: 1.35, stop: 0 #fff, stop: 1 #ddd);
}

QPushButton:pressed {
background: qradialgradient(cx: 0.4, cy: -0.1,
fx: 0.4, fy: -0.1,
radius: 1.35, stop: 0 #fff, stop: 1 #fff);
}
</string>
   </property>
   <property name="title">
    <string>Servo</string>
   </property>
   <widget class="QSlider" name="verticalSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>21</width>
      <height>71</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
      <horstretch>21</horstretch>
      <verstretch>71</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>21</width>
      <height>71</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>21</width>
      <height>71</height>
     </size>
    </property>
    <property name="toolTip">
     <string>Choose potential position for the servo</string>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>97</number>
    </property>
    <property name="sliderPosition">
     <number>49</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
    <property name="invertedAppearance">
     <bool>false</bool>
    </property>
    <property name="invertedControls">
     <bool>false</bool>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksAbove</enum>
    </property>
    <property name="tickInterval">
     <number>10</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spn_value">
    <property name="geometry">
     <rect>
      <x>35</x>
      <y>70</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>42</width>
      <height>22</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>42</width>
      <height>22</height>
     </size>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>97</number>
    </property>
    <property name="value">
     <number>49</number>
    </property>
   </widget>
   <widget class="QCheckBox" name="chk_enable">
    <property name="geometry">
     <rect>
      <x>85</x>
      <y>70</y>
      <width>51</width>
      <height>17</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Select servo to include in next stored position or playback</string>
    </property>
    <property name="text">
     <string>Enable</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_play">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>40</y>
      <width>94</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Play slider position to servo</string>
    </property>
    <property name="text">
     <string>Send</string>
    </property>
    <property name="icon">
     <iconset resource="../SerialPortIcons.qrc">
      <normaloff>:/res/images/play_large_green.png</normaloff>:/res/images/play_large_green.png</iconset>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>20</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Current:</string>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_position">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>20</y>
      <width>21</width>
      <height>16</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Last position sent to this servo</string>
    </property>
    <property name="text">
     <string>97</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources>
  <include location="../SerialPortIcons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>spn_value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>verticalSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>27</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>verticalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spn_value</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>50</x>
     <y>80</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
